From b004aca1f52e059baeec4d0a04854952ae2789a8 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Mon, 27 Nov 2023 21:44:42 +0800
Subject: [PATCH 7/9] get output of ostree-prepare-root back

The LAT requires the output of ostree-prepare-root, but commit
[298c601d ostree-prepare-root: Log journal message after finding
deployment] redirect the output as systemd journal message

Get it back

Upstream-Status: Inappropriate [LAT specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>

Rebased to ostree 2023.7.
Skip unnecessary output as debug

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 src/switchroot/ostree-prepare-root.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/switchroot/ostree-prepare-root.c b/src/switchroot/ostree-prepare-root.c
index 27d06fa..18b8234 100644
--- a/src/switchroot/ostree-prepare-root.c
+++ b/src/switchroot/ostree-prepare-root.c
@@ -157,6 +157,8 @@ resolve_deploy_path (const char *root_mountpoint)
                    SD_ID128_FORMAT_VAL (OSTREE_PREPARE_ROOT_DEPLOYMENT_MSG), "DEPLOYMENT_PATH=%s",
                    resolved_path, "DEPLOYMENT_DEVICE=%" PRIu64, (uint64_t)stbuf.st_dev,
                    "DEPLOYMENT_INODE=%" PRIu64, (uint64_t)stbuf.st_ino, NULL);
+
+  printf ("Resolved OSTree target to: %s\n", deploy_path);
   return deploy_path;
 }
 
@@ -396,7 +398,7 @@ main (int argc, char *argv[])
                  PREPARE_ROOT_CONFIG_PATH);
     }
   const bool sysroot_currently_writable = !path_is_on_readonly_fs (root_arg);
-  g_print ("sysroot.readonly configuration value: %d (fs writable: %d)\n", (int)sysroot_readonly,
+  g_debug ("sysroot.readonly configuration value: %d (fs writable: %d)\n", (int)sysroot_readonly,
            (int)sysroot_currently_writable);
 
   /* Work-around for a kernel bug: for some reason the kernel
@@ -495,7 +497,7 @@ main (int argc, char *argv[])
           g_assert (composefs_config->enabled != OT_TRISTATE_NO);
           if (composefs_config->enabled == OT_TRISTATE_MAYBE && errsv == ENOENT)
             {
-              g_print ("composefs: No image present\n");
+              g_debug ("composefs: No image present\n");
             }
           else
             {
-- 
2.27.0

